### ISSUE TITLE
distsql: Plan and execute joins with interleaved tables

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#1574]
Background: https://airtable.com/tblD3oZPLJgGhCmch/viwdV1sM5KEK2L14K/recUrD0hgoXFHAP2d

PRs:

**Comment by jseldess** [comment]
@andreimatei, once this work is done, please add the `ready` label and links to relevant PRs here so we can figure out whether docs should be updated/expanded.

**Comment by jseldess** [comment]
Duplicate of https://github.com/cockroachdb/docs/issues/2049
